### PR TITLE
Added monad instance, changed functor style

### PR DESCRIPTION
**Pull request by walseb**
https://github.com/ivanperez-keera/dunai/issues/212
So I made use of the `_` wildcard here because the applicative also did. And I made `fmap` infix to match the the other instances more. But I'm not sure if I used the best style here.

### CODE REVIEW

File: examples/bearriver/src/FRP/BearRiver.hs [hunk 1]
```diff
@@ -175,17 +175,20 @@ afterEachCat = afterEachCat' 0
...
 
 instance Applicative Event where
   pure = Event
-
   Event f <*> Event x = Event (f x)
   _       <*> _       = NoEvent
 
+instance Monad Event where
+  return          = pure
+  (Event x) >>= f = f x
```
Comment (turion) [inline examples/bearriver/src/FRP/BearRiver.hs:189]: You can drop the parentheses around `Event x`

File: examples/bearriver/src/FRP/BearRiver.hs [hunk 2]
```diff
@@ -175,17 +175,20 @@ afterEachCat = afterEachCat' 0
 
 
 -- * Events
-
 instance Functor Event where
-  fmap f NoEvent   = NoEvent
-  fmap f (Event c) = Event (f c)
+   f `fmap` (Event c) = Event (f c)
```
Comment (turion) [inline examples/bearriver/src/FRP/BearRiver.hs:179]: I personally find this a bit awkward to read and don't see the advantage.

File: examples/bearriver/src/FRP/BearRiver.hs [hunk 3]
```diff
@@ -175,17 +175,20 @@ afterEachCat = afterEachCat' 0
...
+   _ `fmap` _         = NoEvent
 
 instance Applicative Event where
   pure = Event
-
   Event f <*> Event x = Event (f x)
   _       <*> _       = NoEvent
 
+instance Monad Event where
```
Comment (turion) [inline examples/bearriver/src/FRP/BearRiver.hs:187]: If you want you can add a haddock in the line before that says that this instance is the same as `Maybe`'s.

File: examples/bearriver/src/FRP/BearRiver.hs [hunk 4]
```diff
@@ -175,17 +175,20 @@ afterEachCat = afterEachCat' 0
...
+  return          = pure
+  (Event x) >>= f = f x
+  _         >>= _ = NoEvent
```
Comment (turion) [inline examples/bearriver/src/FRP/BearRiver.hs:190]: I would find it easier to reason about if you mention the pattern here.

**Comment by walseb** [comment]
Thanks! Everything should be fixed now

**Comment by ivanperez-keera** [comment]
Hi, @walseb . Thanks a lot!!

I have merged these commits into the develop branch to be included in the next release.

I made some adaptations, both to the commit message, and to the style. I never really formalized any of those, so they "feel right" and I could argue why I prefer them, but it's more of a soft preference than a hard rule that I'm applying. If you want, take a look :)

As always, commit authorship of the commits from you that I modified (squashed & message reworded) was maintained.

**Comment by ivanperez-keera** [comment]
Thank you also @turion for the comments and contribution :)

**Pull request closed without merging**